app/lib/submission_builder/ty2024/states/md/documents/md502.rb

```diff
@@ -120,6 +120,7 @@ def document
       xml.ExemptionAmount calculated_fields.fetch(:MD502_LINE_19) if calculated_fields.fetch(:MD502_DEDUCTION_METHOD) == "S"
       xml.StateTaxComputation do
         xml.TaxableNetIncome calculated_fields.fetch(:MD502_LINE_20) if calculated_fields.fetch(:MD502_DEDUCTION_METHOD) == "S"
+        add_element_if_present(xml, "StateIncomeTax", :MD502_LINE_21)
```

Comment:
[q] when do we decide to use this method? is it usually indicated by the ticket somewhere?

Reply:
I just use it whenever the minOccurs on the xml element in the XSD is 0 but an equivalent i see others use it `xml.StateIncomeTax calculated_fields.fetch(:MD502_LINE_21) if calculated_fields.fetch(:MD502_LINE_21).present?` and then I think some other forms have some other methods as well

Reply:
This is good to know! Ty